core/src/autogluon/core/trainer/abstract_trainer.py

```diff
@@ -1134,6 +1134,15 @@ def save(self):
         if self.low_memory:
             self.models = models
 
+    def compile_models(self, **kwargs):
+        models_to_compile = list(self.models.keys())
```

Comment:
Don't use self.models. Instead

```
if models == 'all':
    models = self.get_model_names(can_infer=True)
```

Reply:
Done. 

In addition to this, an argument `model_names` has been added with default value to be 'all'.